OsmAnd/src/net/osmand/aidl/OsmandAidlApi.java

```diff
@@ -1967,6 +1966,47 @@ public boolean processResult(GPXFile result) {
 		gpxAsyncLoaderTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
 	}
 
+	private static int partsCounter = 0;
+	private static long startTime = 0;
+
+	boolean appendDataToFile(final String filename, final byte[] data) {
```

Comment:
Add boolean flag to close or not close the file if you are going to keep reopening same file